clients/localdir/client.go

```diff
@@ -116,10 +136,9 @@ func listFiles(clientPath string, predicate func(string) (bool, error)) ([]strin
 // ListFiles implements RepoClient.ListFiles.
 func (client *localDirClient) ListFiles(predicate func(string) (bool, error)) ([]string, error) {
 	client.once.Do(func() {
-		client.files, client.errFiles = listFiles(client.path, predicate)
+		client.files, client.errFiles = listFiles(client.path)
 	})
-
-	return client.files, client.errFiles
+	return applyPredicate(client.path, client.files, client.errFiles, predicate)
```

Comment:
Consider making this a struct fn so you don't have to pass all the `client.*` objects.

Reply:
I removed `client.path` as it was not needed, so it takes only 3 arguments.